### ISSUE TITLE
Sentry configuration should not require Git installed

### ISSUE THREAD

**Issue by RedBeardCode** [painless-software/painless-continuous-delivery#9]
During running of of 

`docker-compose run application python manage.py migrate`

I got the following traceback.

``` bash
/usr/local/lib/python3.5/site-packages/environ/environ.py:617: UserWarning: Error reading /opt/painlesstest/.env - if you're not configuring your environment separately, check this.
  "environment separately, check this." % env_file)
Traceback (most recent call last):
  File "manage.py", line 10, in <module>
    execute_from_command_line(sys.argv)
  File "/usr/local/lib/python3.5/site-packages/django/core/management/__init__.py", line 367, in execute_from_command_line
    utility.execute()
  File "/usr/local/lib/python3.5/site-packages/django/core/management/__init__.py", line 316, in execute
    settings.INSTALLED_APPS
  File "/usr/local/lib/python3.5/site-packages/django/conf/__init__.py", line 53, in __getattr__
    self._setup(name)
  File "/usr/local/lib/python3.5/site-packages/django/conf/__init__.py", line 41, in _setup
    self._wrapped = Settings(settings_module)
  File "/usr/local/lib/python3.5/site-packages/django/conf/__init__.py", line 97, in __init__
    mod = importlib.import_module(self.SETTINGS_MODULE)
  File "/usr/local/lib/python3.5/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 986, in _gcd_import
  File "<frozen importlib._bootstrap>", line 969, in _find_and_load
  File "<frozen importlib._bootstrap>", line 958, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 673, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 665, in exec_module
  File "<frozen importlib._bootstrap>", line 222, in _call_with_frames_removed
  File "/opt/painlesstest/application/settings.py", line 115, in <module>
    'release': raven.fetch_git_sha(dirname(pardir)),
  File "/usr/local/lib/python3.5/site-packages/raven/versioning.py", line 25, in fetch_git_sha
    'Cannot identify HEAD for git repository at %s' % (path,))
raven.exceptions.InvalidGitRepository: Cannot identify HEAD for git repository at
```

Thanks for help in advance.

**Comment by bittner** [comment]
Thanks for reporting!

The exception is related to the [Sentry setup in the Django settings](https://github.com/painless-software/painless-continuous-delivery/blob/master/%7B%7Bcookiecutter.project_slug%7D%7D/_/frameworks/Django/application/settings.py#L128-L132), which is trying to automatically figure out the Git commit hash. This way Sentry can tell you which commit an exception or error belongs to, when it occurs.

The setup is, actually, literally taken from the [Sentry docs](https://docs.sentry.io/clients/python/integrations/django/), which works well when you have Git on your production server (in the end). We need to make this smarter, because in a Docker deployment we shouldn't have Git installed, it's usually not needed.

Proposed Solution
-------------

In the Django settings we need to replace `RAVEN_CONFIG` by ...
``` python
RAVEN_CONFIG = {
    'dsn': env('SENTRY_DSN', default=None),
    'release': env('REVISION', default=None),
}
```
(**NOTE:** Theses changes will make your local configuration work immediately.)

... and put `REVISION` into the server environment by creating a `.env` file, which sets the value to the current Git SHA. This should be done each time we build the Docker `application` image, e.g. with something like that: 
``` bash
ENV_FILE=.env
GIT_SHA=$(git rev-parse HEAD)

touch $ENV_FILE
sed -e '/^REVISION=/d' -i $ENV_FILE
echo "REVISION=$GIT_SHA" >> $ENV_FILE
```

**Comment by RedBeardCode** [comment]
What about implementing it as git hook. So with every commit the hash of the commit will be change in the .env file.

**Comment by bittner** [comment]
A (pre-commit) hook is a nice idea. I've got an eye on [ochtra](https://github.com/kvz/ochtra) for years.

We just need to make sure the we can distribute the feature easily (e.g. no need to run an install command or something similar explicitly on each machine that is cloning the repository).

**Comment by bittner** [comment]
Thinking about it, a pre-commit hook would be completely optional. Where we need the Git SHA is on a deployed system only, and we can generate the required `REVISION=...` line in an `.env` file for that environment in the deployment process.

If you want to place a PR with the solution outlined above I'm happy to include it.

**Comment by bittner** [comment]
Fixed by PR #12.
